downloader/src/main/kotlin/vcs/Subversion.kt

```diff
@@ -0,0 +1,120 @@
+
+                private fun getLineValue(
+                        linePrefix: String) = infoCommandResult.requireSuccess().stdout().lineSequence()
+                        .first { it.startsWith(linePrefix) }.removePrefix(linePrefix).trim()
+            }
+
+    override fun isApplicableProvider(vcsProvider: String) = vcsProvider.toLowerCase() in listOf("subversion", "svn")
+
+    override fun isApplicableUrl(vcsUrl: String) = ProcessCapture("svn", "ls", vcsUrl).exitValue() == 0
+
+    override fun download(vcsUrl: String, vcsRevision: String?, vcsPath: String?, version: String,
+                          targetDir: File): String {
+
+        runSVNCommand(targetDir, "co", vcsUrl, "--depth", "empty", ".")
+
+        val revision = if (vcsRevision != null && vcsRevision.isNotBlank()) {
+            vcsRevision
+        } else {
+            if (version.isNotBlank()) {
+                try {
+                    log.info { "Trying to determine revision for version: $version" }
+                    val tagsList = runSVNCommand(targetDir, "list", "$vcsUrl/tags").stdout().trim().lineSequence()
+                    val tagName = tagsList.firstOrNull {
+                        it.trimEnd('/').endsWith(version)
+                                || it.trimEnd('/').endsWith(version.replace('.', '_'))
+                    }
+
+                    val xml = runSVNCommand(targetDir,
+                                             "log",
+                                             "$vcsUrl/tags/$tagName",
+                                             "--xml").stdout().trim()
+                    val xmlMapper = ObjectMapper(XmlFactory()).registerKotlinModule()
+                    val logEntries: List<SubversionLogEntry> = xmlMapper.readValue(xml, xmlMapper.typeFactory
+                            .constructCollectionType(List::class.java, SubversionLogEntry::class.java))
+                    logEntries.firstOrNull()?.revision ?: ""
+
+                } catch (e: IOException) {
+                    if (Main.stacktrace) {
+                        e.printStackTrace()
+                    }
+
+                    log.warn { "Could not determine revision for version: $version. Falling back to fetch everything" }
+                    ""
+                }
+
+            } else {
+                ""
+            }
+        }
+
+        if (vcsPath != null && vcsPath.isNotBlank()) {
+            targetDir.resolve(vcsPath).mkdirs()
+        }
+
+        if (revision.isNotBlank()) {
+            runSVNCommand(targetDir, "up", "-r", revision, "--set-depth", "infinity", vcsPath?.apply { } ?: "")
+        } else {
+            runSVNCommand(targetDir, "up", "--set-depth", "infinity", vcsPath?.apply { } ?: "")
+        }
+
+        val workDir = Subversion.getWorkingDirectory(targetDir)
+        return workDir.getRevision()
+    }
+
+    private fun runSVNCommand(workingDir: File, vararg args: String) = ProcessCapture(workingDir, "svn", *args).requireSuccess()
```

Comment:
AFAIK the Java style is to write acronyms as camel-case in function names, i.e. use `runSvnCommand`.